api/src/routes/settings.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  Type,
+  type FastifyPluginCallbackTypebox
+} from '@fastify/type-provider-typebox';
+
+export const settingRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.addHook('onRequest', fastify.authenticateSession);
+
+  fastify.put(
+    '/update-my-profileui',
+    {
+      schema: {
+        body: Type.Object({
+          profileUI: Type.Object({
+            isLocked: Type.Boolean(),
+            showAbout: Type.Boolean(),
+            showCerts: Type.Boolean(),
+            showDonation: Type.Boolean(),
+            showHeatMap: Type.Boolean(),
+            showLocation: Type.Boolean(),
+            showName: Type.Boolean(),
+            showPoints: Type.Boolean(),
+            showPortfolio: Type.Boolean(),
+            showTimeLine: Type.Boolean()
+          })
+        }),
+        response: {
+          200: Type.Object({
+            message: Type.String(),
+            type: Type.Literal('success')
```

Comment:
One thing I came across is that these seem to overwrite the potentially intended response. I sent a request and got this response:

```
{
    "message": "flash.wrong-updating",
    "type": "success"
}
```

An error caught below (line 69) sent a response status of 200, and it gave me a 'success' message type - but down on that line, it says to return 'danger'

Reply:
@moT01 with the latest changes, throwing inside the try-catch creates the correct message, type and code.